File: Day 04/class_instance.cpp
#include <iostream>

using namespace std;

class Person {
    public:
        int age;
        Person(int initialAge);
        void amIOld();
        void yearPasses();
};

Person::Person(int initialAge) {
    if (initialAge < 0) {
        cout << "Age is not valid, setting age to 0." << endl;
        this->age = 0;
    } else {
        this->age = initialAge;
    }
}

void Person::amIOld() {
    if (this->age < 13) {
        cout << "You are young." << endl;
    } else if (this->age >= 18) {
        cout << "You are old." << endl;
    } else {
        cout << "You are a teenager." << endl;
    }
}

void Person::yearPasses() {
    this->age++;
}

// Supplied code
int main() {
    int t;
    int age;
    cin >> t;
    for (int i = 0; i < t; i++) {
        cin >> age;
        Person p(age);
        p.amIOld();
        for (int j = 0; j < 3; j++) {
            p.yearPasses();
        }
        p.amIOld();

        cout << endl;
    }
}

File: Learn/30-Days-Of-Code/Day 04/class_instance.py
class Person:
    def __init__(self, initial_age):
        if initial_age < 0:
            print("Age is not valid, setting age to 0.")
            self.age = 0
        else:
            self.age = initial_age

    def amIOld(self):
        if self.age < 13:
            print("You are young.")
        elif self.age >= 13 and self.age < 18:
            print("You are a teenager.")
        else:
            print("You are old.")

    def yearPasses(self):
        self.age += 1


# Supplied code
t = int(input())
for i in range(t):
    age = int(input())
    p = Person(age)
    p.amIOld()
    for j in range(3):
        p.yearPasses()
    p.amIOld()
    print("")
